File: leetcode/0169-majority-element/majorityElement.ts
function majorityElement(nums: number[]): number {
  const numCountMap = new Map<number, number>();
  let maxKeyValue: [number, number] = [-1, -1];

  for (const num of nums) {
    if (numCountMap.has(num)) {
      numCountMap.set(num, numCountMap.get(num)! + 1);
      continue;
    }

    numCountMap.set(num, 1);
  }

  for (const [key, value] of numCountMap) {
    if (value <= maxKeyValue[1]) {
      continue;
    }

    maxKeyValue = [key, value];
  }

  return maxKeyValue[0];
}

export default majorityElement;

File: leetcode/0169-majority-element/majorityElement.go
package leetcode

func majorityElement(nums []int) int {
	var numCountMap = map[int]int{}
	var maxKeyValue = []int{-1, -1}

	for _, num := range nums {

		if count, ok := numCountMap[num]; ok {
			numCountMap[num] = count + 1
			continue
		}

		numCountMap[num] = 1
	}

	for key, value := range numCountMap {
		if value <= maxKeyValue[1] {
			continue
		}

		maxKeyValue = []int{key, value}
	}

	return maxKeyValue[0]
}
